### PR TITLE
Aerolyzer_App 13: Add PostgreSQL backend for user accounts

### PR DESCRIPTION
**Pull request by liusop**
The sign up and logout both work, but we are still working on the login. We also need to rebase settings.py because of a conflict.

### CODE REVIEW

**Comment by liusop** [comment]
@kwhitehall login feature now functional

**Comment by kwhitehall** [comment]
Nice one @liusop and @colliell. Did some testing on my end and got it up and running. 
Please update the README with this PR as well. 
Important things to note are: 
1. Postgresql 9.5.g (min requirement) should be installed and running on port 5432 (provide download link for postgresql)
2. The database in Postgresql should be set up with the settings as outlined in settings.py namely:
NAME, USER and PASSWORD settings. 

EDIT ---
On the home page, the "Let's get started" button takes you to the sign up page, but there is also a sign up button. Would it be prudent to remove the sign up button from that page? To make it intuitive to the user that to sign up is the "Let's get started" button, where as to login one uses the fields at the top? (The other pages should keep both login and sign in at the top). The easier fix so as not to change the header may be to update the text in the button on the home page. Thoughts?
--- END EDIT

Finally on the signup page, please allow for the user to enter credentials OR do not show the login and password box on the sign in page until the person signs up. Currently, if you access the signup page, then enter your details the information is not accepted unless the 'sign up'  was completed. As such it may be desirable to show the first visit to the sign in page without log in credentials. (Any page with login and password box should take info by default). Please see screenshot for further explanation of the issue. Feel free to discuss on Slack channel as well. 

signup

File: Aerolyzer/Aerolyzer/settings.py [hunk 1]
```diff
@@ -76,16 +87,17 @@
...
+# )
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',  # 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
-        'NAME': 'address.db',
-        'USER': '',                      # Not used with sqlite3.
-        'PASSWORD': '',                  # Not used with sqlite3.
-        'HOST': '',                      # Set to empty string for localhost. Not used with sqlite3.
-        'PORT': '',                      # Set to empty string for default. Not used with sqlite3.
+        'ENGINE': 'django.db.backends.postgresql',  # 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
```
Comment (kwhitehall) [inline Aerolyzer/Aerolyzer/settings.py]: please remove these comments (cleaner code).

File: Aerolyzer/app/forms.py [hunk 2]
```diff
@@ -1,23 +1,23 @@
-#files.py
 import re
 from django import forms
 from django.contrib.auth.models import User
 from django.utils.translation import ugettext_lazy as _
- 
+
 class RegistrationForm(forms.Form):
+    username = forms.RegexField(regex=r'^\w+$', widget=forms.TextInput(attrs=dict(required=True, max_length=30)), label=_("Username"), error_messages={ 'invalid': _("Must contain only letters, numbers and underscores. Try again!") })
```
Comment (kwhitehall) [inline Aerolyzer/app/forms.py]: split line. Same for the others. Line length shouldn't exceed 100 characters (as per our Python linting standards)

File: Aerolyzer/app/templates/app/faq.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
 			</div>
 		</div>
 	</div>
-
```
Comment (kwhitehall) [inline Aerolyzer/app/templates/app/faq.html:14]: on line 5, " is missing, should be 
<div class="container" style="height: 375px;">

File: Aerolyzer/app/templates/app/profile.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{% extends "app/base.html" %}
+{% load static %}
+{% block content %}
+
+<header>
+	<div class="container" style="height: 375px;>
```
Comment (kwhitehall) [inline Aerolyzer/app/templates/app/profile.html]: add " to close style

File: Aerolyzer/app/templates/app/upload.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{% extends "app/base.html" %}
+{% load static %}
+{% block content %}
+
+<header>
+	<div class="container" style="height: 375px;>
```
Comment (kwhitehall) [inline Aerolyzer/app/templates/app/upload.html]: add  "

File: Aerolyzer/app/urls.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 from django.conf.urls import url
+from django.contrib.auth import views as auth_views
```
Comment (kwhitehall) [inline Aerolyzer/app/urls.py:3]: remove white space in lib imports

**Review by kwhitehall — DISMISSED** [review]

**Review by kwhitehall — CHANGES_REQUESTED** [review]
This is looking good guys! 
Please address these "linting" issues before merging.

File: Aerolyzer/Aerolyzer/settings.py [hunk 7]
```diff
@@ -11,9 +11,9 @@
 """
 
 import os
-from whitenoise import WhiteNoise
 import django.contrib.auth
 django.contrib.auth.LOGIN_URL = '/'
```
Comment (kwhitehall) [inline Aerolyzer/Aerolyzer/settings.py]: move this line after all the imports

File: Aerolyzer/app/templates/app/base.html [hunk 8]
```diff
@@ -44,7 +44,7 @@
                 <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                     <span class="sr-only">Toggle navigation</span> Menu <i class="fa fa-bars"></i>
                 </button>
-                <a class="navbar-brand page-scroll" href="#page-top">Aerolyzer</a>
+                <a lass="navbar-brand page-scroll" href="#page-top">Aerolyzer</a>
```
Comment (kwhitehall) [inline Aerolyzer/app/templates/app/base.html]: Think this is a typo here.

File: Aerolyzer/app/views.py [hunk 9]
```diff
@@ -22,7 +22,7 @@ def index(request):
 			return HttpResponseRedirect('gallery')
 		else:
 			return render(request, 'app/index.html', {'login_message' : 'That username/password doesn\'t work!'},)
-	return render(request, 'app/index.html', {'user': request.user},)
+	return render(request, 'app/index.html', {'user': request.user, 'is_index':True,},)
```
Comment (kwhitehall) [inline Aerolyzer/app/views.py]: should there be a comma before the brace?

File: README.md [hunk 10]
```diff
@@ -12,9 +12,16 @@ To run the Aerolyzer app locally:
 
 * ```cd Aerolyzer/```
 * ```python manage.py runserver```
-  
+
 * Open your browser to http://127.0.0.1:8000/app/
 
+<<<<<<< HEAD
```
Comment (kwhitehall) [inline README.md]: remove these git conflict comments i.e.  <<<<<<< HEAD and >>>>>>> 8bcb558c2c76cab7363834d64753975e922d55dc

**Review by kwhitehall — CHANGES_REQUESTED** [review]
Thank you for addressing the concerns. A few more are here to be addressed before merging please.

File: README.md [hunk 11]
```diff
@@ -12,9 +12,16 @@ To run the Aerolyzer app locally:
...
+##Important things to note are:
+
+1.Postgresql 9.5.g (min requirement) should be installed and running on port 5432. Postgresql can be downloaded at https://www.postgresql.org/download/
```
Comment (kwhitehall) [inline README.md]: Check PostgreSql version.

File: README.md [hunk 12]
```diff
@@ -12,9 +12,16 @@ To run the Aerolyzer app locally:
...
+2.The database in Postgresql should be set up with the settings as outlined in settings.py namely:
+NAME, USER and PASSWORD settings.
```
Comment (kwhitehall) [inline README.md]: Please indicate what these values should be.

File: Aerolyzer/app/static/js/dropzone.js [hunk 13]
```diff
@@ -0,0 +1,1767 @@
+
```
Comment (kwhitehall) [inline Aerolyzer/app/static/js/dropzone.js]: @liusop  please remove this file from this commit.

**Comment by kwhitehall** [comment]
Thanks! @liusop and @colliell. 
Merging this. Great work! 
Re the dropzone.js if it is to be added to code base, open up an issue for it and add make a separate PR.

**Review by kwhitehall — APPROVED** [review]

**Pull request merged**